packages/api/src/db/setting.ts

```diff
@@ -0,0 +1,47 @@
+export type Setting<T = unknown> = {
+  id: string
+
+  name: string
+  value: T | null
+  settingRestriction?: SettingRestriction
+}
+
+export interface SettingInput<T> {
+  name: string
+  value: T | null
+}
+
+export interface CreateSettingArgs<T> {
+  name: string
+  value: T | null
+  settingRestriction?: SettingRestriction
+}
+
+export interface UpdateSettingArgs {
+  id: string
+  input: SettingInput<any>
+}
+
+export interface DeleteSettingArgs {
+  id: string
+}
+
+export interface SettingRestriction {
```

Comment:
I had backend in mind here because I felt frontend validation shouldn't be enough for changing setting values, is this correct? I have a check in the mutation file.